api/src/main/java/ca/bc/gov/educ/api/batchgraduation/processor/RunCertificateRegenerationProcessor.java

```diff
@@ -25,27 +30,43 @@ public class RunCertificateRegenerationProcessor implements ItemProcessor<UUID,
 	Long batchId;
 
 	@Override
-	public Integer process(UUID key) throws Exception {
-		GraduationStudentRecordDistribution stuRec = restUtils.getStudentData(key.toString());
-		if (stuRec != null) {
-			LOGGER.info("Processing partitionData: studentID = {}", stuRec.getStudentID());
+	@Nullable
+	public Integer process(@NotNull StudentCredentialDistribution item) throws Exception {
+		if (item.getStudentID() != null) {
+			LOGGER.info("Processing partitionData: studentID = {}", item.getStudentID());
 			summaryDTO.setBatchId(batchId);
 			try {
+				if (StringUtils.isBlank(item.getPen())) {
+					item.setPen(getPenNumber(item.getStudentID()));
+				}
 				summaryDTO.setProcessedCount(summaryDTO.getProcessedCount() + 1L);
-				Integer count = restUtils.runRegenerateStudentCertificate(stuRec.getPen());
+				Integer count = restUtils.runRegenerateStudentCertificate(item.getPen());
 				if (count > 0) {
-					restUtils.updateStudentGradRecord(key, batchId, "CERTREGEN");
+					restUtils.updateStudentGradRecord(item.getStudentID(), batchId, "CERTREGEN");
 				}
 				return count;
 			} catch(Exception e) {
 				LOGGER.error("Unexpected Error: {}", e.getLocalizedMessage());
-				summaryDTO.updateError(key,"GRAD-GRADUATION-API IS DOWN","Graduation API is unavailable at this moment");
+				summaryDTO.updateError(item.getStudentID(),"GRAD-GRADUATION-API IS DOWN","Graduation API is unavailable at this moment");
```

Comment:
We are logging the error, but in summaryDTO, it just says GRAD-GRADUATION-API IS DOWN which is not accurate. If we need this error in the DTO, please update with thrown error message.

Reply:
Updated the message title properly, and put the thrown error message as message description into the DTO.